components/retro-go/rg_gui.c

```diff
@@ -265,89 +264,47 @@ static size_t get_glyph(uint32_t *output, const rg_font_t *font, int points, int
-                    output[y] = pattern[y];
-            }
-            else
-            {
-                uint8_t *pattern = (uint8_t *)font->data + (c * font->height);
-                for (int y = 0; y < font->height; y++)
-                    output[y] = pattern[y];
-            }
-        }
-    }
-    else if (font->type == 1) // Proportional
-    {
-        // Based on code by Boris Lovosevic (https://github.com/loboris)
-        int charCode, adjYOffset, width, height, xOffset, xDelta;
-        const uint8_t *data = font->data;
-        while (1)
-        {
-            charCode = *data++;
-            adjYOffset = *data++;
-            width = *data++;
-            height = *data++;
-            xOffset = *data++;
-            xOffset = xOffset < 0x80 ? xOffset : -(0xFF - xOffset);
-            xDelta = *data++;
-
-            if (charCode == c || charCode == 0xFF)
-                break;
+    uint16_t bitmap_index;
+    uint8_t xDelta, width, height;
+    int8_t xOffset, yOffset;
 
-            if (width != 0)
-                data += (((width * height) - 1) / 8) + 1;
-        }
+    const uint8_t *bitmap_data = font->bitmap_data;
+    const rg_font_glyph_dsc_t *glyph_dsc = font->glyph_dsc;
+    
+    // getting the glyph informations :
+    bitmap_index = glyph_dsc[c].bitmap_index;
+    xDelta = glyph_dsc[c].adv_w;
+    width = glyph_dsc[c].box_w;
+    height = glyph_dsc[c].box_h;
+    xOffset = glyph_dsc[c].ofs_x;
+    yOffset = glyph_dsc[c].ofs_y-2;
 
-        // If the glyph is not found, we fallback to the basic font which has most glyphs.
-        // It will be ugly, but at least the letter won't be missing...
-        if (charCode != c)
-            return get_glyph(output, &font_basic8x8, RG_MAX(8, points - 2), c);
+    const uint8_t *data = &bitmap_data[bitmap_index];
 
-        glyph_width = RG_MAX(width, xDelta);
-        if (output)
+    size_t glyph_width = xDelta;
+    if (output)
+    {
+        int ch = 0, mask = 0x80;
+        for (int y = 0; y < height; y++)
         {
-            int ch = 0, mask = 0x80;
-            for (int y = 0; y < height; y++)
+            output[yOffset + y] = 0;
+            for (int x = 0; x < width; x++)
             {
-                output[adjYOffset + y] = 0;
-                for (int x = 0; x < width; x++)
+                if (((x + (y * width)) % 8) == 0)
                 {
-                    if (((x + (y * width)) % 8) == 0)
-                    {
-                        mask = 0x80;
-                        ch = *data++;
-                    }
-                    if ((ch & mask) != 0)
-                        output[adjYOffset + y] |= (1 << (xOffset + x));
-                    mask >>= 1;
+                    mask = 0x80;
+                    ch = *data++;
                 }
+                if ((ch & mask) != 0)
+                    output[yOffset + y] |= (1 << (xOffset + x));
+                mask >>= 1;
             }
         }
     }
 
-    // Vertical stretching
```

Comment:
Note that this block isn't just used by the (now removed) basic renderer, it's also used by the `RG_TEXT_BIGGER` flag which doubles the height of any font. So it should be preserved :)